src/Nethermind/Nethermind.Optimism/OptimismLegacyTxDecoder.cs

```diff
@@ -0,0 +1,39 @@
+// SPDX-FileCopyrightText: 2024 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System;
+using Nethermind.Core;
+using Nethermind.Core.Crypto;
+using Nethermind.Core.Specs;
+using Nethermind.Serialization.Rlp;
+using Nethermind.Serialization.Rlp.TxDecoders;
+using Nethermind.TxPool;
+
+namespace Nethermind.Optimism;
+
+public class OptimismLegacyTxDecoder : LegacyTxDecoder<Transaction>
+{
+    protected override Signature? DecodeSignature(ulong v, ReadOnlySpan<byte> rBytes, ReadOnlySpan<byte> sBytes, Signature? fallbackSignature = null,
+        RlpBehaviors rlpBehaviors = RlpBehaviors.None)
+    {
+        if (v == 0 && rBytes.IsEmpty && sBytes.IsEmpty)
+        {
+            return null;
+        }
+        return base.DecodeSignature(v, rBytes, sBytes, fallbackSignature, rlpBehaviors);
+    }
+}
+
+public class OptimismLegacyTxValidator : ITxValidator
+{
+    public ValidationResult IsWellFormed(Transaction transaction, IReleaseSpec releaseSpec)
+    {
+        // In op bedrock eip1559 activated with bedrock
+        if (releaseSpec.IsEip1559Enabled)
+        {
+            return transaction.Signature is null ? new ValidationResult("Empty signature") : ValidationResult.Success;
```

Comment:
Before we were not performing any validations on Deposit transactions, but now we require that post-Bedrock transactions contain a non-null signature. Is this correct? If so, shouldn't we also validate the signature (using `SignatureTxValidator`)?

Reply:
if signature is null here it means that while decoding signature in `DecodeSignature` we got inside  `if (v == 0 && rBytes.IsEmpty && sBytes.IsEmpty)`. Without this pr we would crash inside `DecodeSignature`.

Reply:
That makes sense, what I'm missing is why we don't validate the signature of a Legacy transaction post bedrock. More so, why are we not performing the standard Legacy validations:

https://github.com/NethermindEth/nethermind/blob/d00410fbba8dd79d287c428c9aa4b02f399f9737/src/Nethermind/Nethermind.Consensus/Validators/TxValidator.cs#L24-L30

With these changes, Legacy transactions that are processed in Optimism are not validated at all, when previously they went through several checks.

Reply:
Oh, that means I made a mistake. But why we don't check DepositTransaction?
`api.RegisterTxType<OptimismTransactionForRpc>(new OptimismTxDecoder<Transaction>(), Always.Valid);`

Reply:
It dates back to the original Optimism plugin PR where the `OptimismTxValidator` performed no checks on Deposit transactions

https://github.com/NethermindEth/nethermind/blob/6edd397acfea5f262796db5a0f15592cf5d6cf53/src/Nethermind/Nethermind.Optimism/OptimismTxValidator.cs#L20-L26

We didn't change the logic in the Transaction refactoring, so the current validator for Deposit transactions always succeeds.

Reply:
Addressed in https://github.com/NethermindEth/nethermind/pull/7887.